tls/s2n_connection.c

```diff
@@ -560,6 +560,8 @@ int s2n_connection_wipe(struct s2n_connection *conn)
 #pragma GCC diagnostic pop
 #endif
 
+    /* Remove parsed extensions array from client_hello beforing zeroing connection */
+    GUARD(s2n_client_hello_free_parsed_extensions(&conn->client_hello));
```

Comment:
nitpick: could we move this (freeing the extensions) before the cloning? for example right after ```s2n_free(&conn->status_response)```? That way it would be grouped with the other freeing/cleaning statements that prepare the connection for the save-zero-recover block.